platform/linux/mod.rs

```diff
@@ -200,27 +201,33 @@ impl UnixSender {
                                          cmsg_buffer.offset(1) as *mut _ as *mut c_int,
                                          fds.len());
 
-                // Put this on the heap so address remains stable across function return.
-                let mut iovec = Box::new(iovec {
+                let iovec = iovec {
                     iov_base: data_buffer.as_ptr() as *const c_char as *mut c_char,
                     iov_len: data_buffer.len(),
-                });
-
+                };
                 let msghdr = msghdr {
                     msg_name: ptr::null_mut(),
                     msg_namelen: 0,
-                    msg_iov: &mut *iovec,
+                    msg_iov: &iovec as *const _ as *mut _,
```

Comment:
The idea was that `sendmsg()` only reads data, rather than mutating, and thus it makes sense to also make the `iovec` immutable here.

Later I realised that the `iovec` itself (as opposed to the buffers it references) is never mutated in the `recvmsg()` either -- and indeed it's declared as `const` in the definition of the same structure in libc. So I considered adjusting it on the receiver side too -- but ultimately decided not to bother, as one of the things I intend to do next is switching to the libc definitions anyway.

Maybe I should add this cleanup here after all, to avoid confusion...